StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1137ElementsShouldHaveTheSameIndentation.cs

```diff
@@ -0,0 +1,432 @@
+                break;
+
+            case SyntaxKind.Parameter:
+                elements.AddRange(((ParameterSyntax)member).AttributeLists);
+                break;
+
+            default:
+                break;
+            }
+
+            elements.Add(member);
+        }
+
+        private static void CheckElements<T>(SyntaxNodeAnalysisContext context, SyntaxList<T> elements)
+            where T : SyntaxNode
+        {
+            if (elements.Count < 2)
+            {
+                return;
+            }
+
+            CheckElements(context, elements.ToImmutableList());
+        }
+
+        private static void CheckElements<T>(SyntaxNodeAnalysisContext context, SeparatedSyntaxList<T> elements)
+            where T : SyntaxNode
+        {
+            if (elements.Count < 2)
+            {
+                return;
+            }
+
+            CheckElements(context, elements.ToImmutableList());
+        }
+
+        private static void CheckElements<T>(SyntaxNodeAnalysisContext context, ImmutableList<T> elements)
+            where T : SyntaxNode
+        {
+            if (elements.Count < 2)
+            {
+                return;
+            }
+
+            elements = elements.RemoveAll(
+                element =>
+                {
+                    SyntaxToken firstToken = GetFirstTokenForAnalysis(element);
+                    return firstToken.IsMissingOrDefault() || !firstToken.IsFirstInLine();
+                });
+
+            if (elements.Count < 2)
+            {
+                return;
+            }
+
+            // Try to reorder the list so the first item is not an attribute list. This element will establish the
+            // expected indentation for the entire collection.
+            int desiredFirst = elements.FindIndex(x => !x.IsKind(SyntaxKind.AttributeList));
+            if (desiredFirst > 0)
+            {
+                T newFirstElement = elements[desiredFirst];
+                elements = elements.RemoveAt(desiredFirst).Insert(0, newFirstElement);
+            }
+
+            bool first = true;
+            string expectedIndentation = null;
+            foreach (T element in elements)
+            {
+                SyntaxToken firstToken = GetFirstTokenForAnalysis(element);
+                SyntaxTrivia indentationTrivia = firstToken.LeadingTrivia.LastOrDefault();
+                string indentation = indentationTrivia.IsKind(SyntaxKind.WhitespaceTrivia) ? indentationTrivia.ToString() : string.Empty;
+
+                if (first)
+                {
+                    expectedIndentation = indentation;
+                    first = false;
+                    continue;
+                }
+
+                if (string.Equals(expectedIndentation, indentation, StringComparison.Ordinal))
+                {
+                    // This handles the case where elements are indented properly
+                    continue;
+                }
```

Comment:
:question: What is the expected behavior if a mixed set of tabs and spaces is used?
I would expected that `<TAB><TAB><SPC> ` would be equal to `<TAB><SPC><SPC><SPC>` if indentation and tab size are both 2, otherwise it may be hard to explain the difference to users.